src/mscorlib/src/System/Enum.cs

```diff
@@ -727,6 +663,51 @@ internal unsafe Object GetValue()
             }
         }
 
+        [System.Security.SecuritySafeCritical]
+        private unsafe ulong GetValueAsULong()
+        {
+            fixed (void* pValue = &JitHelpers.GetPinningHelper(this).m_data)
+            {
+                unchecked
+                {
+                    switch (InternalGetCorElementType())
+                    {
+                        case CorElementType.I1:
+                            return (ulong)*(sbyte*)pValue;
+                        case CorElementType.U1:
+                            return (ulong)*(byte*)pValue;
+                        case CorElementType.Boolean:
+                            return Convert.ToUInt64(*(bool*)pValue, CultureInfo.InvariantCulture);
+                        case CorElementType.I2:
+                            return (ulong)*(short*)pValue;
+                        case CorElementType.U2:
+                            return (ulong)*(ushort*)pValue;
+                        case CorElementType.Char:
+                            return (ulong)*(char*)pValue;
+                        case CorElementType.I4:
+                            return (ulong)*(int*)pValue;
+                        case CorElementType.U4:
+                            return (ulong)*(uint*)pValue;
+                        case CorElementType.R4:
+                            return (ulong)*(float*)pValue;
+                        case CorElementType.I8:
+                            return (ulong)*(long*)pValue;
+                        case CorElementType.U8:
+                            return (ulong)*(ulong*)pValue;
+                        case CorElementType.R8:
+                            return (ulong)*(double*)pValue;
```

Comment:
Is this case right for float/double based enums?

Float/double based enums are gray area. Some things do not quite make sense or work well for them, but they work fine for the most part.